File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpContainerTakeBatchRequest.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpContainerTakeBatchRequest(
        List<Move> moves,
        boolean dryRun,
        boolean stopOnError
) {
    public record Move(
            RMcpContainerMoveBatchRequest.Endpoint from,
            Integer toInventorySlot,
            int count
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpContainerTakeBatchRequest.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpContainerTakeBatchRequest(
    val moves: List<Move>,
    val dryRun: Boolean,
    val stopOnError: Boolean
) {
    @JvmRecord
    data class Move(
        val from: RMcpContainerMoveBatchRequest.Endpoint,
        val toInventorySlot: Int,
        val count: Int
    )
}
